src/components/AddCompanies/AddCompanies.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Form, FormGroup, ControlLabel, FormControl, Col } from 'react-bootstrap';
+import './styles.less';
+
+const propTypes = {
+    addCompany: PropTypes.func.isRequired,
+};
+
+class AddCompanies extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			name: '',
+			address: '',
+			phone: '',
+			revenue: '',
+		};
+
+		this.addNewCompany = this.addNewCompany.bind(this);
+		this.handleFormControlChange = this.handleFormControlChange.bind(this);
+	}
```

Comment:
Here I'm binding the class functions I'll use to ensure the context of the specific instance of the classes is referenced... otherwise you will have a this equal to undefined. 
You can also use Arrow Functions to implicitly bind... but I wanted to show you this way because this can cause headaches if you are new to React.